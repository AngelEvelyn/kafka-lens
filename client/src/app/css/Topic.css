.topic {
  border: 1px solid rgb(185, 183, 183);
  width: 50%;
  border-radius: 5px;
  justify-self: center;
  align-self: center;
  padding: 20px;
}
