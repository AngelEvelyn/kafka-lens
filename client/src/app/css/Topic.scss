@import 'theme';

.topic-header {
    font-size: 21px;
    cursor: pointer;
}

.topic-padding {
    padding-left: 56px;
}

.topic-padding:hover, .topic-padding:focus {
    background-color: $highlight-bar-color;
    cursor: pointer;
}