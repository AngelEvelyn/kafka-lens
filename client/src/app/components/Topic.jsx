import React from 'react';
import '../css/Topic.css';

// const Topic = props => {
//   console.log('be able to see props: ', props)
//   function showPartitions(e) {
//     console.log('show e.target: ', e.target)
//     console.log('show props: ', props)
//   }
//   return (
//     <div className="topic" topicinfo={props.topicInfo} onClick={showPartitions}>
//       <div>
//         <span>Topic Name: {props.topicInfo.topic}</span> <br />
//         <span>Partitions: {props.topicInfo.partition}</span>
//       </div>
//       <br />
//     </div>
//   );
// };

// export default Topic;

class Topic extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      topics: []
    };

    // this.clickTopic = this.clickTopic.bind(this);
    // this.showPartitions = this.showPartitions.bind(this);
  }
  // Lifecycle methods
  componentDidMount() {
    //code here
  }
  // Methods
  // showPartitions(event) {
  //   //code here
  //   // console.log('firing showPartitions function')
  // }
  render() {

    return (
      <div className="topic">
        <div>
          <span>Topic Name: {this.props.topicInfo.topic}</span> <br />
          <span>Partitions: {this.props.topicInfo.partition}</span>
        </div>
          <button id={this.props.id} onClick={this.props.showPartitions}>View</button>
        <br />
      </div>
    );
  }
}

export default Topic;

