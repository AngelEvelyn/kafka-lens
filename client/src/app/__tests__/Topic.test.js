import React from 'react';
import { shallow } from 'enzyme';

import Topic from '../components/Topic';

describe('Topic.js unit tests', () => {
  let wrapper;
  const props = {
    id: 1,
    showPartitions: jest.fn(),
    topicInfo: {
      topic: 'test1',
      partition: 5
    }
  };
  beforeAll(() => {
    wrapper = shallow(<Topic {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call showPartitions on click', () => {
    wrapper
      .find('button')
      .at(0)
      .simulate('click');
    expect(props.showPartitions).toHaveBeenCalled();
  });
});
