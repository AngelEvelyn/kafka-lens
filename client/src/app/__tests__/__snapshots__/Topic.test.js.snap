// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic.js unit tests Should render without errors 1`] = `
<div
  className="topic-header"
>
  <div
    id={1}
    onClick={[MockFunction]}
  >
    test1
  </div>
</div>
`;
